### PR TITLE
Remove unnecessary modifiers from interfaces

### PR DESCRIPTION
**Pull request by yanlyub**
There are still lots of unnecessary `public` modifiers in almost all interfaces, even though everything is `public` by default in an interface, anything else would make no sense, an interface is a contract with the outside world and private/protected/package-visibility members don't belong in an interface.

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
    */
   public VisibilityCondition getAttributeVisibility(String name);
 
-  public static class Util {
+  public class Util {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java:76]: Won't this change make the inner class depend on the outer class when it doesn't need to? Is that binary compatible?

Comment (yanlyub) [reply vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java:76]: There is no outer class, it's an interface, completely different thing. Classes inside interfaces are by default static, they cannot be non-static. This is binary compatible, this is not even a change, just removal of redundant information.

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java:76]: Ah. I didn't have enough context above this to see that the containing block was an interface.

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java:76]: Looks like all the others are also that way.

Comment (yanlyub) [reply vassal-app/src/main/java/VASSAL/build/AutoConfigurable.java:76]: Yes I'm too missing a feature that lets me open the whole file from the PR view, I need it often and it's very inconvenient to open the PR and the changed file locally or navigate to it's version on GitHub in another browser tab.

**Pull request merged**